cses/dp/Counting_Tilings.cpp

```diff
@@ -157,6 +157,15 @@ Mint dp[2][1 << 10];
 
 int main(){
     int n = scd(), m = scd();
+
+    // Adding an optimization :  when n and m are both odd , then ans is always 0 for 2x1 tilling 
+
+    if( (n&1) && (m&1))
```

Comment:
Looks good to merge.
One nit - let's do early return inside this 'if' block not use else.
Also keep the opening bracket in same line as 'if' to match the file style.
Will merge after that.

Reply:
Ok brother , I have made the changes as you have mentioned.